File: luogu/p1734.py
n = int(input())

a = [0]*1001
f = [0]*1001
for i in range(1, 1001):
    for j in range(i+i, 1001, i):
        a[j] += i
for i in range(1, 1001):
    for j in range(n, i-1, -1):
        f[j] = max(f[j], f[j-i]+a[i])
print(f[n])
        

File: luogu/p1734.cpp
#include<bits/stdc++.h>

using namespace std;

int a[1001], f[1001], n;
void pre()
{
    for(int i = 1; i < 1001; i++)
        for(int j = i+i; j < 1001; j+=i)
            a[j] += i;
}
int main()
{
    pre();
    cin >> n;
    for(int i = 1; i <= 1000; i++)
    {
        for(int j = n; j >= i; j--)
            f[j] = max(f[j], f[j-i]+a[i]);
    }
    cout << f[n] << endl;
    return 0;
}